src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -558,6 +556,21 @@ public void saveSubscriptions(String subscriptionsJson, boolean reconcileCapacit
     }
   }
 
+  @Transactional
+  private void determineSubscriptionOffering(
+      org.candlepin.subscriptions.db.model.Subscription subscription) {
+    // should look up the offering and set it before additional processing
+    var offer =
+        offeringRepository.findOfferingBySku(
+            subscription.getSubscriptionProductIds().stream().findFirst().orElse(null));
```

Comment:
I dislike `findFirst` because it's just silent if the set has more than one element (which we wouldn't be expecting here).  If we use `MoreCollectors.toOptional()`, it will throw an exception if there is more than one element in the set.
```
            subscription.getSubscriptionProductIds().stream()
                .collect(MoreCollectors.toOptional())
                .orElse(null));
```